syntax = "proto3";
package api.v1;

import "google/api/annotations.proto";

option go_package = "/api";

service ConfigService {
  rpc GetConfig(GetConfigRequest) returns (GetConfigResponse) {
    option (google.api.http) = {get: "/config"};
  }

  rpc UpdateConfig(UpdateConfigRequest) returns (UpdateConfigResponse) {
    option (google.api.http) = {
      post: "/config:update"
      body: "*"
    };
  }

  rpc GetIpBlackList(GetIpBlackListRequest) returns (GetIpBlackListResponse) {
    option (google.api.http) = {get: "/config/iplist"};
  }

  rpc UpdateIpBlackList(UpdateIpBlackListRequest) returns (UpdateIpBlackListResponse) {
    option (google.api.http) = {
      post: "/config/iplist:update"
      body: "*"
    };
  }
}

message GetConfigRequest {}

message GetConfigResponse {
  bool http_compression = 1;
}

message UpdateConfigRequest {
  repeated string path = 1;
  bool http_compression = 2;
  bool https = 3;
}

message UpdateConfigResponse {
  bool http_compression = 1;
  bool https = 2;
}

message GetIpBlackListRequest {}

message GetIpBlackListResponse {
  repeated string ip_list = 1;
}

message UpdateIpBlackListRequest {
  enum ActionType {
    ACTION_TYPE_UNSPECIFIED = 0;
    ACTION_TYPE_ADD = 1;
    ACTION_TYPE_REMOVE = 2;
  }
  ActionType action = 1;
  repeated string ip_list = 2;
}

message UpdateIpBlackListResponse {}
